tests/D2L.CodeStyle.Analyzers.Test/Specs/ReflectionSerializerConstructorAnalyzer.cs

```diff
@@ -0,0 +1,241 @@
+						Value = value;
+					}
+					public int Value { get; }
+					// ↓↓↓
+					WithImplicittInternalConstructor() { }
+				}
+
+				[ReflectionSerializer]
+				public sealed class WithExplicitInternalConstructor {
+					public WithExplicitInternalConstructor( int value ) {
+						Value = value;
+					}
+					public int Value { get; }
+					// ↓↓↓
+					internal WithExplicitInternalConstructor() { }
+				}
+
+				[ReflectionSerializer]
+				public sealed class WithProtectedConstructor {
+					public WithProtectedConstructor( int value ) {
+						Value = value;
+					}
+					public int Value { get; }
+					// ↓↓↓
+					protected WithProtectedConstructor() { }
+				}
+
+				[ReflectionSerializer]
+				public sealed class WithPrivateConstructor {
+					public WithPrivateConstructor( int value ) {
+						Value = value;
+					}
+					public int Value { get; }
+					// ↓↓↓
+					private WithPrivateConstructor() { }
+				}
+			}
+		}
+
+		namespace PropertyStyle {
+
+			namespace DefaultConstructor {
+
+				[ReflectionSerializer]
+				public sealed class ImplicitConstructor {
+					public int Value { get; set; }
+				}
+
+				[ReflectionSerializer]
+				public sealed class ImplicitConstructor_WithStaticConstructor {
+					public int Value { get; set; }
+					// ↓↓↓
+					static ImplicitConstructor_WithStaticConstructor() { }
+				}
+			}
+
+			namespace EmptyConstructor {
+
+				[ReflectionSerializer]
+				public sealed class Explicit {
+					// ↓↓↓
+					public Explicit() { }
+					public int Value { get; set; }
+				}
+
+				[ReflectionSerializer]
+				public sealed class WithStaticConstructor {
+					public WithStaticConstructor() { }
+					public int Value { get; set; }
+					// ↓↓↓
+					static WithStaticConstructor() { }
+				}
+
+				[ReflectionSerializer]
+				public sealed class WithImplicitInternalConstructor {
+					public WithImplicitInternalConstructor() { }
+					public int Value { get; set; }
+					// ↓↓↓
+					WithImplicitInternalConstructor( int value ) { }
+				}
+
+				[ReflectionSerializer]
+				public sealed class WithInternalConstructor {
+					public WithInternalConstructor() { }
+					public int Value { get; set; }
+					// ↓↓↓
+					internal WithInternalConstructor( int value ) { }
+				}
+
+				[ReflectionSerializer]
+				public sealed class WithProtectedConstructor {
+					public WithProtectedConstructor() { }
+					public int Value { get; set; }
+					// ↓↓↓
+					protected WithProtectedConstructor( int value ) { }
+				}
+
+				[ReflectionSerializer]
+				public sealed class WithPrivateConstructor {
+					public WithPrivateConstructor() { }
+					public int Value { get; set; }
+					// ↓↓↓
+					private WithPrivateConstructor( int value ) { }
+				}
+			}
+		}
+
+		namespace MultiplePublicConstructors {
+
+			/* ReflectionSerializer_MultiplePublicConstructors */ [ReflectionSerializer]
+			public sealed class EmptyAndNonEmpty {
+				public EmptyAndNonEmpty() { }
+				public EmptyAndNonEmpty( int arg0 ) { }
+			} /**/

```

Comment:
This is OK but it might be a bit nicer to make the error message less red, e.g. highlight the second constructor in the clase, or the class name in the no constructors case...

It'd not a huge deal, but in a bigger class it looks weird to have the whole thing highlighted. It's not so bad with these errors though because the error message has an obvious fix.